index.js

```diff
@@ -402,6 +393,10 @@
   var Inconsistent =
   new _Type(INCONSISTENT, '', '', always2('???'), K(false), [], {});
 
+  //  NoArguments :: Type
+  var NoArguments =
+  new _Type(NO_ARGUMENTS, '', '', always2('()'), K(true), [], {});
```

Comment:
`NoArguments` and `Unit` represent related but distinct ideas.

Were we to define `const Unit = {};` and the accompanying Type value, `f ()` and `f (Unit)` would still be different expressions: one applies `f` to no arguments; the other applies `f` to one argument.

It could be confusing to name the above Type value `Unit` as we may one day wish to define a Type value equivalent to Haskell's `()`. This would logically be named `Unit`.

Reply:
In Haskell `()` is a nullary tuple. I like to think that the only difference between Haskell and JavaScript, is that JavaScript forces you to pattern-match your tuples when passing them around, and you can only instantiate them at call time.

```js
//    f :: Tuple a b -> c -> ...
//      -- alernatively written as
//    f :: (a, b) -> c -> ...
const f = (a, b) => c => ...
```

When viewing it like this, Unit has no addressable value level member (only at call time).